prow/plank/controller.go

```diff
@@ -415,9 +415,11 @@ func (c *Controller) syncPendingJob(pj prowapi.ProwJob, pm map[string]corev1.Pod
 				}
 				break
 			}
-			// Pod is running. Do nothing.
-			c.incrementNumPendingJobs(pj.Spec.Job)
-			return nil
+			if pod.DeletionTimestamp == nil {
```

Comment:
Is deleted different from an eviction?

Reply:
Yes, an eviction is indicated in the pod status and does not result in the pod object being deleted.

The only case I am aware of where a cluster could delete a pod is cluster autoscaler downscaling. To prevent that, we have a PDB in openshift (I'll create a PR to add this to the config here as well, because its often used as a reference).

So far, we went by the assumption that if a pod vanishes, this means we have to just re-create it and are good. Since "pod vanishes" is not an expected condition and indicates that someone or something messed with the pods, I would like to make this an error condition, because it can easily hide issues. Should we in the future find other cases where we expect a pod to get deleted by some other entity, we can try to handle those explicitly.